### PR TITLE
Clean up some magic values and refactor a bit

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/editor.c [hunk 1]
```diff
@@ -304,7 +304,7 @@ static int editor_read_s6(const char *path)
 	if (rw != NULL) {
 		if (!sawyercoding_validate_checksum(rw)) {
 			SDL_RWclose(rw);
-			RCT2_GLOBAL(RCT2_ADDRESS_ERROR_TYPE, uint8) = 255;
+			RCT2_GLOBAL(RCT2_ADDRESS_ERROR_TYPE, uint8) = ERROR_TYPE_FILE_LOAD;
 			RCT2_GLOBAL(RCT2_ADDRESS_ERROR_STRING_ID, uint16) = STR_FILE_CONTAINS_INVALID_DATA;
```
Comment (IntelOrca) [inline src/editor.c]: For your next commit, you could also integrate these ```RCT2_ADDRESS_ERROR_TYPE``` and ```RCT2_ADDRESS_ERROR_STRING_ID``` variables.

File: src/localisation/string_ids.h [hunk 2]
```diff
@@ -1614,6 +1614,7 @@ enum {
 
 	STR_THIS_RIDE_CANNOT_BE_MODIFIED = 3046,
 	STR_LOCAL_AUTHORITY_FORBIDS_DEMOLITION_OR_MODIFICATIONS_TO_THIS_RIDE = 3047,
+	MARKETING_CAMPAIGNS_FORBIDDEN_BY_LOCAL_AUTHORITY = 3048,
```
Comment (IntelOrca) [inline src/localisation/string_ids.h]: You forgot to prefix this with ```STR_```

File: src/ride/ride.c [hunk 3]
```diff
@@ -6013,7 +6013,7 @@ money32 ride_create(int type, int subType, int flags, int *outRideIndex, int *ou
 		}
 
 		if (ride->type == RIDE_TYPE_TOILETS) {
-			if (RCT2_GLOBAL(0x01358838, uint32) & (1 << 31)) {
+			if (RCT2_GLOBAL(RCT2_ADDRESS_SAME_PRICE_THROUGHOUT, uint32) & (1 << 31)) {
```
Comment (IntelOrca) [inline src/ride/ride.c]: ```shop_item_has_common_price(SHOP_ITEM_ADMISSION)```

File: src/ride/ride.h [hunk 4]
```diff
@@ -841,7 +841,9 @@ enum {
 	SHOP_ITEM_EMPTY_JUICE_CUP,
 	SHOP_ITEM_ROAST_SAUSAGE,
 	SHOP_ITEM_EMPTY_BOWL_BLUE,
-	SHOP_ITEM_COUNT = 56
+	SHOP_ITEM_COUNT = 56,
+	SHOP_ITEM_ADMISSION = 31,
```
Comment (IntelOrca) [inline src/ride/ride.h]: Might as well move this to inbetween ```SHOP_ITEM_EMPTY_BOTTLE``` and ```SHOP_ITEM_PHOTO2```

**Comment by Gymnasiast** [comment]
Ok, that should address everything.

**Pull request merged**